### ISSUE TITLE
CUDA opt-level is wrongly capped to 2 on MSVC tool family

### ISSUE THREAD

**Issue by KarelPeeters** [alexcrichton/cc-rs#679]
When compiling `.cuda(true)` code on an otherwise MSVC toolchain, when passing the opt-level this piece of code runs: https://github.com/alexcrichton/cc-rs/blob/fba7feded71ee4f63cfe885673ead6d7b4f2f454/src/lib.rs#L1499-L1505

But nvcc does support opt-level 3, it's even the default: https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#ptxas-options-opt-level

This may not be that important, in my experiments I don't really see any performance difference between levels 2 and 3, but this does seem like a small oversight that slipped in.

I'm not sure what the best fix is, maybe just adding a new `ToolFamily` for NVCC. I'm not sure if that works with the other flags though.

**Comment by dot-asm** [comment]
Quoting `nvcc --help`
```
--optimize <level>                              (-O)                            
        Specify optimization level for host code.
```
Note "host." The provided reference suggests that you're talking about the device code, but `-O` has no effect on it. It's still the default 3, and if you wanted to change that, then you'd have to pass the corresponding ptxas-specific option through `-Xptxas` nvcc option.

Fun fact about `-O3` permitted by nvcc driver. If you run `nvcc -O3 -keep ...` on your Windows system and examine .res files, you'll find that the nvcc driver passes `-O2` to the vendor compiler, cl. This is because cl doesn't have `-O3`, `-O2` is the highest.

To summarize, there is no need to do anything, everything works as intended.

**Comment by KarelPeeters** [comment]
Ah well spotted, it seems I confused myself a bit there.
